### PR TITLE
Added Documentation on Generic Reordering Support in Extending Section

### PR DESCRIPTION
**Pull request by Coder-Ankit001**
### What does this change?
Improves the documentation for generic reordering in `ModelViewSet`, with clearer examples and guidance on preparing existing data when enabling `sort_order_field`.

- Added a section explaining reordering requirements
- Included an example for handling existing records

### Why ?
The current docs mention reordering briefly, but do not include practical examples or use-case guidance, and they do not cover the null or unpopulated sort_order scenario from https://github.com/wagtail/wagtail/pull/13485. This adds a small documentation section with working examples and notes on relevant edge cases.

### Tests
Not applicable (documentation-only change)

Fixes https://github.com/wagtail/wagtail/pull/13615

### CODE REVIEW

**Comment by Coder-Ankit001** [comment]
Hi all, just checking in on this PR to see if there’s any feedback or if anything else is needed from my side.
This update improves the docs on generic reordering with practical examples as discussed in #13615.
Thanks for your time!

File: docs/extending/generic_views.md [hunk 1]
```diff
@@ -95,6 +95,84 @@ If you would like to make further customizations to the filtering mechanism, you
...
+
+```python 
+# Example data migration function
+
+def populate_sort_order(apps, schema_editor):
+    Employee = apps.get_model('myapp', 'Employee')
+    for i, employee in enumerate(Employee.objects.order_by('id')):
+        employee.sort_order = i
+        employee.save()
+```
```
Comment (gasman) [inline docs/extending/generic_views.md:187]: I think it would be useful to show this in the context of the data migration - it might not be obvious to everyone that this function should be hooked up with `RunPython`:

```suggestion
```python 
# Example data migration

from django.db import migrations, models


def populate_sort_order(apps, schema_editor):
    Employee = apps.get_model('myapp', 'Employee')
    for i, employee in enumerate(Employee.objects.order_by('id')):
        employee.sort_order = i
        employee.save()
        
class Migration(migrations.Migration):
    dependencies = [
        # ...
    ]

    operations = [
        migrations.RunPython(
            populate_sort_order,
            migrations.RunPython.noop,
        ),
    ]
```
```

(looks like the Github suggestions interface might have cut off the ending ``` here - please make sure that's gone through properly!)

**Comment by Coder-Ankit001** [comment]
Thanks @gasman! I've updated it to show the full migration context with RunPython as suggested.

**Comment by gasman** [comment]
Thanks @Coder-Ankit001! Merged in 03a4f5b3eb1dc41f42274f96b0606aef5f21f491 (main) and 49765d9e77 (stable/7.3.x).

File: docs/extending/generic_views.md [hunk 2]
```diff
@@ -85,12 +85,107 @@ The {attr}`~ModelViewSet.list_display` attribute can be set to specify the colum
...
+This viewset can then be registered with the Wagtail admin to make it available under the URL `/admin/employee/`, by adding the following to `wagtail_hooks.py`:
+
+```python
+from wagtail import hooks
+from .views import employee_viewset
+
+@hooks.register("register_admin_viewset")
+def register_viewset():
+    return employee_viewset
+```
```
Comment (laymonage) [inline docs/extending/generic_views.md:153]: I don't understand why this needs to be in the docs about reordering support. We already have an example on how the viewset is registered at the main `ModelViewSet` section just a few sections above, and this sentence and example is pretty much copypasted from there. This reordering section should only add the details necessary to enable the reordering feature.

File: docs/extending/generic_views.md [hunk 3]
```diff
@@ -85,12 +85,107 @@ The {attr}`~ModelViewSet.list_display` attribute can be set to specify the colum
 
 If the model has an integer field for custom ordering, you can set the name of that field as the {attr}`~ModelViewSet.sort_order_field` attribute on the viewset or on the model. This will add a "Sort item order" button on the listing view, allowing users to reorder the items by dragging and dropping them.
 
+```{versionadded} 7.2
+The ability to reorder items on the listing view was added in Wagtail 7.2.
+```
```
Comment (laymonage) [inline docs/extending/generic_views.md:90]: This paragraph is now redundant, and the `versionadded` note should be moved to the new section.
